compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/codegen/rvalue.rs

```diff
@@ -716,26 +716,24 @@ impl<'tcx> GotocCtx<'tcx> {
         ty: Ty<'tcx>,
         trait_ty: &'tcx ty::TyS<'tcx>,
     ) -> Expr {
-        let drop_instance = Instance::resolve_drop_in_place(self.tcx, ty);
+        let drop_instance = Instance::resolve_drop_in_place(self.tcx, ty).polymorphize(self.tcx);
         let drop_sym_name = self.symbol_name(drop_instance);
-        let param_types = vec![self.codegen_ty(trait_ty)];
 
         // The drop instance has the concrete object type, for consistency with
         // type codegen we need the trait type for the function parameter.
-        let trait_fn_ty =
-            Type::code_with_unnamed_parameters(param_types, Type::unit()).to_pointer();
+        let trait_fn_ty = self.trait_vtable_drop_type(trait_ty);
 
         if let Some(drop_sym) = self.symbol_table.lookup(&drop_sym_name) {
             Expr::symbol_expression(drop_sym_name, drop_sym.clone().typ)
                 .address_of()
                 .cast_to(trait_fn_ty)
         } else {
-            // TODO: check why in https://github.com/model-checking/rmc/issues/66
-            debug!(
-                "WARNING: drop_in_place not found for {:?}",
-                self.readable_instance_name(drop_instance),
-            );
-            Type::void_pointer().null().cast_to(trait_fn_ty)
+            self.codegen_unimplemented(
```

Comment:
Does this typecheck? We want a pointer to an unimplemented function, and instead we're getting an unimplemented function that returns a fn pointer.

Reply:
The issue is that this code is called in a constructor, so it will always fail at CBMC time (which is sound, but not helpful).  I think what we want is to make a function whose body is `unimplemented`, and put a pointer to that function into the vtable.